### ISSUE TITLE
Look into enabling configuring the cp-ksql-cli docker image through environment variables

### ISSUE THREAD

**Issue by apurvam** [confluentinc/ksql#1361]
Currently, the `cp-ksql-cli` docker image can only be configured by mounting a config file in the image and then passing it to the command.

We should look into seeing if we can support configuring the cli through environment variables, as we can with the `cp-ksql-server`. This comment https://github.com/confluentinc/ksql/pull/1347#issuecomment-393730858 has some context about why we think this can't be done today.

**Comment by miguno** [comment]
**Suggestion 1:** You [wrote in the other thread](https://github.com/confluentinc/ksql/pull/1347#issuecomment-393736485):

> The problem with emulating the existing CMD ["/etc/confluent/docker/run"] ultimately ending up with exec <service start command> is that the exec is really final. You cannot append arguments to it.

I think what you mean is that our helper scripts like [cp-ksql-server/include/etc/confluent/docker/run](https://github.com/confluentinc/ksql/blob/master/cp-ksql-server/include/etc/confluent/docker/run) end with the following lines:

```bash
echo "===> Launching ... "
exec /etc/confluent/docker/launch
```

Why can't we modify this to the following in order to pass command line arguments?

```bash
echo "===> Launching ... "
exec /etc/confluent/docker/launch $@
```

In other words, there is not inherent limitation for bash's `exec` that prevents us from passing further command line arguments to it.  It's simply that our *current* utility scripts don't do that.

**Suggestion 2:** We could update our `ksql` script to directly parse any defined `KSQL_<var>` env variables such as `KSQL_BOOTSTRAP_SERVERS`, instead of doing this indirectly (as we do for the server image) by having our Docker tooling create a KSQL config file from the `KSQL_*` env vars, then launching the KSQL process with the generated config file.

**Comment by apurvam** [comment]
Thanks @miguno 

I think suggestion 2 will definitely work, but is more involved.

As for suggestion 1, I think that would work as well when used with the docker `ENTRYPOINT` instruction (our existing scripts use `CMD`). With `ENTRYPOINT` any additional arguments get appended to the default, in this case `/etc/confluent/docker/run`. There is no reason why those can't be propagated to the actual command.  

With `CMD`, any additional arguments overwrite the argument to `CMD`. 

The cli image already uses `ENTRYPOINT`, but I never thought to use it in this way. It seems promising.

**Comment by miguno** [comment]
Is this still an issue, @apurvam ?
